codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -165,11 +175,31 @@
          * <li>The connection is marked as going away.</li>
          * </ul>
          * <p>
-         * The caller is expected to {@link Http2Stream#open(boolean)} the stream.
+         * If the stream is intended to initialized to {@link Http2Stream.State#OPEN} then use
+         * {@link #createStream(int, boolean)} otherwise optimizations in {@link Listener}s may not work
+         * and memory may be thrashed. The caller is expected to {@link Http2Stream#open(boolean)} the stream.
+         * @param streamId The ID of the stream
+         * @see Http2Stream#open(boolean)
+         */
+        Http2Stream createIdleStream(int streamId) throws Http2Exception;
```

Comment:
I don't really see any advantage to combining creation and opening.  I think I prefer the interface as it was.